sktime/forecasting/base/adapters/_neuralforecast.py

```diff
@@ -134,16 +136,49 @@ def algorithm_parameters(self: "_NeuralForecastAdapter") -> dict:
         - custom model name (``alias``) - used from ``algorithm_name``
         """
 
+    def _get_valid_parameters(self: "_NeuralForecastAdapter") -> dict:
+        """Get valid parameters for the underlying NeuralForecast algorithm class.
+
+        Returns
+        -------
+        dict
+            valid list of arguments for the underlying algorithm class
+        """
+        from pytorch_lightning import Trainer
+
+        model_class = self.algorithm_class
+        trainer_params = list(signature(Trainer.__init__).parameters.keys())
+        valid_parameters = list(signature(model_class.__init__).parameters.keys())
+        valid_parameters += trainer_params
+
+        sktime_parameters = self.algorithm_parameters
+
+        for sktime_param in list(sktime_parameters.keys()):
+            if sktime_param not in valid_parameters:
+                if sktime_parameters[sktime_param] is not None:
```

Comment:
Can you please explain why "not None" is required, and why should we not warn always?

Are you intending this to be a check for non-default value passed by users? Does all parameters have None as default?

Reply:
if suppose the user is on version `1.6.4` and didn't pass the optimizer still he will get a warning that optimizer is being dropped as top level interfaces pass optimizer param through `self.algorithm_params`. Now the default value of optimizer is None so it will have no effect on the model, but suppose the value of optmizer by is passed by user or is optim object by default  then showing a warning that it is being dropped would be good as it will have effect on the model, while params with None won't have any effect. So basically this will log warning only when user passes a invalid parameter with actual value or the default value of the parameter will affect the model

Reply:
> Are you intending this to be a check for non-default value passed by users? Does all parameters have None as default?

Are you certain that all newly added parameters in `neuralforecast` in future releases will have default value of `None`? Just for illustration purposes, what happens if they add `n_jobs=-1`?

Will it not be better to compare against the default values of sktime, e.g. using `self.get_default_params`? I'll suggest to check the `statsforecast` changes by @arnaujc91 for reference.

Reply:
Yes it will be to better to compare with default values